client/my-sites/themes/themes-site-selector-modal.jsx

```diff
@@ -140,11 +145,14 @@ class ThemesSiteSelectorModal extends React.Component {
 	}
 }
 
+const bindGetSiteSlug = state => siteId => getSiteSlug( state, siteId );
+const bindGetWpcomTheme = state => themeId => getTheme( state, 'wpcom', themeId );
+
 export default connect( state => ( {
 	// We don't need a <QueryTheme /> component to fetch data for the theme since the
 	// ThemesSiteSelectorModal will always be called from a context where those data are available.
 	// FIXME: Since the siteId and themeId are part of the component's internal state, we can't use them
 	// here. Instead, we have to return helper functions.
-	getSiteSlug: siteId => getSiteSlug( state, siteId ),
-	getWpcomTheme: themeId => getTheme( state, 'wpcom', themeId ),
+	getSiteSlug: bindGetSiteSlug( state ),
+	getWpcomTheme: bindGetWpcomTheme( state ),
```

Comment:
do you also need to mark this as impure? These functions will no longer change on every run of mapStateToProps, so rendering may get skipped?

Reply:
They will change on every run and will return a different function object on each call. We'll have unneeded rerenders on every `state` change, but that was the previous behavior, too. Not easy to fix and out of scope for this PR.